### PR TITLE
Bad code -> Good code

### PR DESCRIPTION
**Pull request by Sibi-Agilan**

### CODE REVIEW

**Comment by Atharv-Agarwal** [comment]
Honestly I dont like the 3rd commit, starts with is not a great way to manage code

**Comment by Sibi-Agilan** [comment]
Lots of small commands are the bad way to code with dpy, it takes around 0.2 to 0.3 seconds to register a command. The startswith method is used to shorten the code and makes it much more readable. Repeating a block of code more than once in python means you're doing something wrong. Try to implement the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle in your code.

**Review by Atharv-Agarwal — DISMISSED** [review]

File: autoreact/autoreact.py [hunk 1]
```diff
@@ -1,55 +1,53 @@
...
+		await ctx.send(f"Deleted reaction {reaction1} for {user_id}")
+
+	@commands.Cog.listener()
+	async def on_message(self, message):
+		if message.author.bot:
+			return
+		for x in message.mentions:
+			uid = await self.coll.find_one({"user_id": str(x.id)})  # getting the user ID if in db then getting reaction
+			if not uid:
+				return
```
Comment (Atharv-Agarwal) [inline autoreact/autoreact.py:35]: Should there be a extra indent here?

**Review by Atharv-Agarwal — CHANGES_REQUESTED** [review]

**Review by Atharv-Agarwal — CHANGES_REQUESTED** [review]
This needs to be undone since the way my hosting manages the bot is it reboots it when it crashes. Therefore client.close() acts as a way to reboot. Deprecation warning has been fixed as per #2

File: autoreact/autoreact.py (hunk 1 above)
Comment (Sibi-Agilan) [reply autoreact/autoreact.py:35]: Its not needed, my mistake

Comment (Atharv-Agarwal) [reply autoreact/autoreact.py:35]: Could you fix those so I can accept this PR? Thanks for helping

File: discordyt/discordyt.py [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
-    
-    @commands.command()
-    @commands.is_owner()
-    async def startyt(self, ctx):
-        link = await self.togetherControl.create_link(ctx.author.voice.channel.id, 'youtube')
-        await ctx.send(f"Click the blue link!\n{link}")
+class DiscordYT(commands.Cog):
+
+	def __init__(self, client):
+		self.client = client
```
Comment (Atharv-Agarwal) [inline discordyt/discordyt.py:8]: Could you convert this to bot please

File: extras/extras.py [hunk 3]
```diff
@@ -2,125 +2,116 @@
...
+				await asyncio.sleep(5)
+		except ValueError:
+			await ctx.message.reply('You must enter a number!')
+
+	@staticmethod
+	@commands.command()
+	@checks.has_permissions(PermissionLevel.MODERATOR)
+	async def raw(ctx, msg: discord.Message = None):
+		await ctx.send(
+			f"``` {msg.embeds[0].description} ```" if msg and msg.embeds else "Please provide a valid message link / "
```
Comment (Atharv-Agarwal) [inline extras/extras.py:106]: This takes int as a input i.e message ID. Could you change this to say "Please provide a valid message ID."

File: newcomers/newcomers.py [hunk 4]
```diff
@@ -9,78 +9,77 @@
 time_units = {'s': 'seconds', 'm': 'minutes', 'h': 'hours', 'd': 'days', 'w': 'weeks'}
 
 
-class newcomers(commands.Cog):
-    def __init__(self, bot):
-        self.bot = bot
-        self.coll = bot.plugin_db.get_partition(self)
-        self.checker.start()
+def to_seconds(s):
```
Comment (Atharv-Agarwal) [inline newcomers/newcomers.py:12]: Should the definition be outside the class?

File: premiumsupport/premiumsupport.py [hunk 5]
```diff
@@ -68,6 +68,17 @@ class Msg:
                 id = initial_message.id
                 attachments = []
                 stickers = []
+
+            await thread.send(Msg, destination=recipient, from_mod=True, anonymous=True)
```
Comment (Atharv-Agarwal) [inline premiumsupport/premiumsupport.py:72]: This has been updated in the base to check for msg and ping, please update this

File: snipe/snipe.py [hunk 6]
```diff
@@ -1,204 +1,184 @@
...
 
 this_file_directory = Path(__file__).parent.resolve()
 other_file = this_file_directory / "nosnipe.txt"
+
 with open(other_file, "r+") as file:
-    nosnipe = [nosnipe.strip().lower() for nosnipe in file.readlines()]
+	nosnipe = [nosnipe.strip().lower() for nosnipe in file.readlines()]
+
+
+def check_view_perms(channel, member):
```
Comment (Atharv-Agarwal) [inline snipe/snipe.py:13]: Again, should the definition be outside the class?

**Review by Atharv-Agarwal — CHANGES_REQUESTED** [review]
If you could do the changes requested and clarify I can merge this!

File: premiumsupport/premiumsupport.py (hunk 5 above)
Comment (Sibi-Agilan) [reply premiumsupport/premiumsupport.py]: Done

**Comment by Atharv-Agarwal** [comment]
Code broke :(( 
Bfping.py
`if message.author.bot or not message.channel.permissions_for(message.author, discord.Permissions.manage_messages):
TypeError: permissions_for() takes 2 positional arguments but 3 were given` ( Reverted)

extras.py raw was a static method and ctx.send doesnt work. (Reverted)

**Pull request merged**